### DISCUSSION TITLE
Using the Media Library?

### DISCUSSION THREAD

**Discussion by thechristoph** [typemill/typemill#432] [Q&A]
I have just started playing around with Typemill 2. My first impression is that it's the first flat CMS I've tried that is actually simple to set up and configure, so that is very nice. Though I worry that the navigation section can become overgrown very quickly (I'm going to try to use it for a portfolio of sorts as well as writing). I look forward to using it more and learning about it. 

One thing I cannot figure out at all is how to use the Media Library. I can see that if I am making a content page, I can upload an image, and that image becomes part of the Media Library. But is there any way to upload images to the Media Library directly? To me it seems the best way is to upload all your images first to the Media Library, then as you write content, you reference the Media Library to insert your image.  Am I thinking about this all wrong?

**Comment by trendschau** [comment]
Thank you for the feedback! Regarding the navigation: You can collapse and expand the navigation and just open the tree that you are currently working on. I hope this feature in Typemill V2 will also improve the management of larger websites. 

The media lib right now works like you described: You can use it when you create content. In most cases this should be the most efficient way because you work with image where you need them. A bulk upload is a nice idea, I will add it to the backlog and think about it. Images are processed on upload (resized, optionally converted into webp, and more), so I think the amount of images for a one time upload will be limited. 

There are more plans for the media-library, for example search images that are not in use to cleanup the media library. So I will probably plan a dedicated release that improves media handling in general in the future.

**Reply by kentchan4** [reply]
Hello and I just discover this topic, I am having a need for a batch upload image as well.
Other than batch upload, please also consider Media's subfolder / directories. I using typemill for my personal own blog, to record my game play, hobbies and most important is my newly born daughter. I will like to see if media library can have their own "sub-folder" for the need to differeiential different topics :)
